flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/FunctionCatalog.java

```diff
@@ -569,6 +581,8 @@ private Optional<ContextResolvedFunction> resolvePreciseFunctionReference(Object
         CatalogFunction potentialResult = tempCatalogFunctions.get(normalizedIdentifier);
 
         if (potentialResult != null) {
+            registerFunctionJarResources(
+                    oi.asSummaryString(), potentialResult.getFunctionResources());
```

Comment:
Not quite familiar with this component. Could you help me to understand why we need to register resource here?